src/network/networkd-manager.c

```diff
@@ -1469,6 +1468,10 @@ void manager_free(Manager *m) {
         m->rules_foreign = set_free_with_destructor(m->rules_foreign, routing_policy_rule_free);
         set_free_with_destructor(m->rules_saved, routing_policy_rule_free);
 
+        sd_netlink_unref(m->rtnl);
+        sd_netlink_unref(m->genl);
+        sd_resolve_unref(m->resolve);
```

Comment:
`Link` has a reference to `Network` and `Network` has a reference to `NetDev`, But both reference does not increase the reference counter. So, these are freed in the following order:
1. `Link` objects
2. `Network` objects
3. `NetDev` objects
But `rtnl` may has references to `Link`. I have not checked all paths, but the error seems caused when `Link` objects freed. So, I think the best place to call `unref()`s is just after the following line:
```
m->links = hashmap_free_with_destructor(m->links, link_unref);
```

Reply:
That is the assumption and there is a race which can be clearly seen by the valgrind.